src_c/display.c

```diff
@@ -1208,6 +1218,13 @@ pg_set_mode(PyObject *self, PyObject *arg, PyObject *kwds)
                 }
             }
 
+            // SDL doesn't preserve window position in fullscreen mode
+            // However, windows coming out of fullscreen need these to go back into the correct position
+            if (sdl_flags & (SDL_WINDOW_FULLSCREEN | SDL_WINDOW_FULLSCREEN_DESKTOP)) {
```

Comment:
What about switching from fullscreen to fullscreen?

Reply:
It seems to preserve the information through multiple fullscreen set_mode calls in a row. (If that's what you're asking).

However I just realized if you start in fullscreen mode and exit, you will still go to the top corner, which is annoying. I thought I had that working, maybe the MAC changes broke that.